File: go/dsa/data-structures/graph/graph.go
func dfs(nodeId string, visited map[string]bool) { // @viz:def
    visited[nodeId] = true // @viz:mark
    process(nodeId) // @viz:visit

    for _, neighbor := range adj[nodeId] { // @viz:for
        if !visited[neighbor] { // @viz:check
            dfs(neighbor, visited) // @viz:dfs
        }
    }
}

File: cpp/dsa/data-structures/graph/graph.cpp
void dfs(string nodeId, unordered_set<string>& visited) { // @viz:def
    visited.insert(nodeId); // @viz:mark
    process(nodeId); // @viz:visit

    for (string neighbor : adj[nodeId]) { // @viz:for
        if (visited.find(neighbor) == visited.end()) { // @viz:check
            dfs(neighbor, visited); // @viz:dfs
        }
    }
}